src/basic/capability-util.c

```diff
@@ -370,3 +370,18 @@ int drop_capability(cap_value_t cv) {
 
         return 0;
 }
+
+bool ambient_capabilities_supported(void) {
+        static int cache = -1;
+
+        if (cache >= 0)
+                return cache;
+
+        /* If PR_CAP_AMBIENT returns something valid, or an unexpected error code we assume that ambient caps are
+         * available. */
+
+        cache = prctl(PR_CAP_AMBIENT, PR_CAP_AMBIENT_IS_SET, CAP_KILL, 0, 0) >= 0 ||
+                !IN_SET(errno, EINVAL, EOPNOTSUPP, ENOSYS);
```

Comment:
Yeah, I guess so. But tje mam page does not explain well the situation such that it returns e.g. EOPNOTSUPP. The reason I made the question is on the sysytem whose kernel does not support kernel prctl() may returns EINVAL. I so not know it ia true or not. But at least I rwad the man page, I underatand that it returns EINVAL in such a situation. If it ia true, then the second condition causes missdetection of if the kernel support ambient capabilities.